pkg/segment/results/mresults/tests/metricsresults_test.go

```diff
@@ -959,3 +959,211 @@ func test_GetResults_Ops(t *testing.T, initialEntries map[uint32]float64, ansMap
 		}
 	}
 }
+
+func Test_GetResults_And(t *testing.T) {
+	results := make(map[string]map[uint32]float64)
+	// There is the expected result: contains only one matching label set
+	results["test.metric.1{color:red,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+	// These two label sets are for metric1 and metric2 respectively.
+	labelStrs1 := []string{"{color:yellow,type:compact,", "{color:red,type:compact,"}
+	labelStrs2 := []string{"{color:red,type:compact,"}
+	test_GetResults_LogicalAndVectorMatchingOps(t,
+		map[uint32]float64{
+			0:    45,
+			7200: 1045,
+		},
+		map[uint32]float64{
+			0:    2,
+			7200: 6,
+		},
+		labelStrs1, labelStrs2, results,
+		[]structs.QueryArithmetic{
+			{
+				LHS:       1,
+				RHS:       2,
+				Operation: utils.LetAnd,
+			},
+		},
+		structs.Downsampler{
+			Interval:   2,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggregation{AggregatorFunction: utils.Avg},
+		},
+	)
+}
+
+func Test_GetResults_Or(t *testing.T) {
+	results := make(map[string]map[uint32]float64)
+	// There is the expected result: contains all unique label sets
+	results["test.metric.1{color:red,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+
+	results["test.metric.1{color:yellow,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+
+	labelStrs1 := []string{"{color:yellow,type:compact,", "{color:red,type:compact,"}
+	labelStrs2 := []string{"{color:red,type:compact,"}
+
+	test_GetResults_LogicalAndVectorMatchingOps(t,
+		map[uint32]float64{
+			0:    45,
+			7200: 1045,
+		},
+		map[uint32]float64{
+			0:    2,
+			7200: 6,
+		},
+		labelStrs1, labelStrs2, results,
+		[]structs.QueryArithmetic{
+			{
+				LHS:       1,
+				RHS:       2,
+				Operation: utils.LetOr,
+			},
+		},
+		structs.Downsampler{
+			Interval:   2,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggregation{AggregatorFunction: utils.Avg},
+		},
+	)
+}
```

Comment:
I think this PR can get merged like this (just address my other nit first)

But later in a different PR it might be good to refactor this to something like the following (not exactly what I have though)
```
func Test_GetResults_Or(t *testing.T) {
	metric1Series1 := "test.metric.1{color:red,type:compact,"
	metric1Series2 := "test.metric.1{color:yellow,type:compact,"
	metric2Series1 := "test.metric.2{color:red,type:compact,"

	// The n-th element contains the metric values for the n-th time-window.
	metric1Series1Values := [][]float64{
		{45},
		{1045},
	}
	metric1Series2Values := [][]float64{
		{2},
		{8, 10},
	}
	metric2Series1Values := [][]float64{
		{},
		{4, 6},
		{17},
	}


	expectedResults := make(map[string][][]float64)
	expectedResults[metric1Series1] = [][]float64{
		{45},
		{1045},
		{17},
	}
	expectedResults[metric1Series2] = [][]float64{
		{2},
		{8, 10},
	}

	helperTestFunction(t, metric1Series1, metric1Series2, metric2Series1,
		metric1Series1Values, metric1Series2Values, metric2Series1Values,
		expectedResults, utils.LetOr)
}
```
(I'm not sure my expected results are correct)

This way I've abstracted away some things that don't really matter for what we're trying to test (e.g., the downsampler), though that would still have to be handled in the helper function. I think something like this will help clarify exactly what the unit test is testing, and make it easier for others to add similar unit tests for whatever features they add

Reply:
Yes, i will refactor that test.